TestVectors/eddsa-2022/proofConfigDataInt.json

```diff
@@ -0,0 +1,11 @@
+{
+  "type": "DataIntegrityProof",
+  "cryptosuite": "eddsa-2022",
+  "created": "2023-02-24T23:36:38Z",
+  "verificationMethod": "https://vc.example/issuers/5678#z6MkrJVnaZkeFzdQyMZu1cgjg7k1pZZ6pvBQ7XJPt4swbTQ2",
+  "proofPurpose": "assertionMethod",
+  "@context": [
+    "https://www.w3.org/ns/credentials/v2",
+    "https://www.w3.org/ns/credentials/examples/v2"
```

Comment:
```suggestion
```

Reply:
Again, not needed... this context is 404, and it defines no terms in the proof.

Reply:
Two separate issues here. The context being 404 and the not a `@context` value defining terms not in the proof configuration.

When I produced this set of test vectors both contexts "https://www.w3.org/ns/credentials/examples/v2", and "https://www.w3.org/ns/credentials/v2" were undefined. Now only the first is undefined. These `@context` values were recommended by @msporny as being forward looking and up to date (too up to date?). I'm assuming the first one will be updated/published soon.

Regarding the `@context` for `proofConfig` the current specification leaves out the canonicalization step for `proofConfig`. To do this step ("Universal RDF Dataset Canonicalization") `@context` needs to be applied to the generated `proofConfig`. I took a robust approach of copying the `@context` from the unsigned document to the `proofConfig` during test vector generation.

However, in the JCS case I did not do this since it seemed unnecessary (JCS canonicalization doesn't use context), but this raised issues with those test vectors from @dlongley for not adding the `@context` to proof config.

Hence I think we need to refine the specification on `proofConfig` generation and canonicalization.

This does show the power of test vectors in helping move the specification along! Cheers Greg

Reply:
> When I produced this set of test vectors both contexts "https://www.w3.org/ns/credentials/examples/v2", and "https://www.w3.org/ns/credentials/v2" were undefined. Now only the first is undefined. These @context values were recommended by @msporny as being forward looking and up to date (too up to date?). I'm assuming the first one will be updated/published soon.

@Wind4Greg https://www.w3.org/ns/credentials/v2 is now live and points to the most up-to-date JSON-LD context for the VCDM. It also includes an extra `@vocab` to https://www.w3.org/ns/credentials/issuer-dependent# which is also live and replaces the usage of an extra context file at https://www.w3.org/ns/credentials/examples/v2 (which does not, and will not, be valid and will stay 404). Ie, the references to the latter should be removed from all the examples.

Reply:
> (which does not, and will not, be valid and will stay 404). Ie, the references to the latter should be removed from all the examples.

Wait, no, big -1 on not defining a v2 examples context.

I am suggesting that we do the following in the spec:

1. Have a section that outlines "How to get started", that shows examples w/ no secondary context.
2. Clearly articulate that the ideal case is that people define clear semantics in their VCs via a secondary context and use that... and then the rest of the specification uses real world examples that use a secondary context, or the examples context (where we need a carefully crafted example).

Reply:
> I am suggesting that we do the following in the spec:
> 
> 1. Have a section that outlines "How to get started", that shows examples w/ no secondary context.
> 2. Clearly articulate that the ideal case is that people define clear semantics in their VCs via a secondary context and use that... and then the rest of the specification uses real world examples that use a secondary context, or the examples context (where we need a carefully crafted example).

I am not against that but, if we do, we should have _something_ at the receiving end of the URL. I am not saying we should have a formal vocabulary, but at least a landing page, akin to what we now at the `.../issuer-dependent` URL.

Reply:
ahh, yes... totally agree... maybe  `@vocab`: `https://example.edu/vocab#` if the examples are exclusive to education tech.

Reply:
Ok, finally found these set of comments. I'm on board w/ what @OR13 and @iherman are suggesting above... once we have that in VCDM, we can use the v2 examples context (not bound to any market vertical... using something like: `"@vocab": "https://www.w3.org/credentials/examples#"` and then that points to a document, much like the issuer-dependent one, but clearly articulates that the vocabulary is only to be used for examples.

Reply:
yep, I don't think W3C should be hosting example context files... This would be an acceptable solution:

```json
"@context": [
    "https://www.w3.org/ns/credentials/v2",
    "https://www.w3.org/ns/credentials/examples/v2"
```

changed to:

```json
"@context": [
    "https://www.w3.org/ns/credentials/v2",
    { "@vocab": "https://www.w3.org/credentials/examples#" }
```

And then all the terms would result in definitions that are just as "defined" as this:

```json
"@context": [
    "https://www.w3.org/ns/credentials/v2"
```

(In the sense that thee term IRIs don't dereference to anything useful and the URL communicates that is going to be the case).

Reply:
Looks like @msporny and @OR13 are starting to converge? What would really be helpful is to agree on the contents of the unsigned document (example document) that is input to the process, i.e., `TestVectors/unsigned.json`.

Cheers Greg

Reply:
Of course, using `@vocab` at all here means that implementations that want to prohibit usage of `@vocab` will have to build special flags to enable its use when generating test vectors. That seems unfortunate, but doable.